.circleci/build_image.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+set -xe
+
+if [[ $# -ne 2 ]]; then
+  echo "Usage: $0 DOCKERFILE IMAGE_NAME"
+  echo ".dockerfile is appended to DOCKERFILE."
+  echo "janitortechnology/ is prepended to IMAGE_NAME."
+  exit 2
+fi
+
+DIR=`dirname $1`
+DOCKERFILE=`basename $1`
+IMAGE_NAME=$2
+
+if [[ -z $CIRCLE_PULL_REQUEST ]]; then
+  # Log in to Docker Hub.
+  docker login -u $DOCKER_USER -p $DOCKER_PASS
+  # Decide whether Docker Hub images should be tagged ":branch-Y" or ":latest".
+  if [[ $CIRCLE_BRANCH != "master" ]]; then
+    DOCKERHUB_TAG="branch-$CIRCLE_BRANCH"
+  else
+    DOCKERHUB_TAG="latest"
+  fi
+else
+  DOCKERHUB_TAG="pr"
```

Comment:
Nit: Please completely remove the `DOCKERHUB_TAG` variable. I think we should either publish images as `:latest` (i.e. the default value, so we can omit the tag) when we're on `master` branch, or not publish them at all.

Reply:
I don't agree with that. There are cases where we want to test new images, such as images utilizing Docker volumes.

Reply:
I have briefly updated things, but the image save/load depends on `DOCKERHUB_TAG` so it's blocked on this discussion regarding `DOCKERHUB_TAG`.